# Hypothesis testing 

Often the confirmation of some testing or alternative hypothesis $H_1$ is a relative affair, where it is measured against some null hypothesis $H_0$. 

* If an alternative hypothesis states that there is some significant relationship between two variables, then the null hypothesis states there is no relationship. 


# p-values 

The p-value quantifies the probability that observed results are the result of random chance. 

If the null hypothesis is true and there is no significant correlation between the population variables X and Y, then any correlation between X and Y observed in our sample would have to be the result of random chance. 

# Statistical testing: Example

Suppose we are told that African elephants have weights distributed normally around a mean of 9000 lbs, with a standard deviation of 900 lbs. 

Pachyderm Adventures has recently measured the weights of 25 African elephants in Gabon and has calculated their average weight as 8637 lbs. 

They claim that these statistics on the Gabonese elephants are significant. Are they correct? 

## What's the null hypothesis in this problem?  What is the alternative hypothesis? 

$H_0$: The average weight of Gabonese elephants is the same as the other elephants' average weight. 

$H_1$: The average weight of Gabonese elephants is not the same as the other elephants' average weight. 

## What kind of statistical test should we perform here? 

> Two-tailed one-sample z-test
* One-sample z-test: We are comparing the mean of a set of measurements to a given population mean and population standard deviation
* Two-tailed: our alternative hypothesis is that the average weight of Gabonese elephants is not the same as the other elephants' weights; this means we want to test is the average weight is smaller or larger than the other elephants' average weight. 

## Compute the test statistic

In [2]:
import numpy as np

In [3]:
x_bar = 8637
n = 25
mu = 9000
sigma = 900

z_statistic = (x_bar - mu)/(sigma/np.sqrt(n))

In [4]:
z_statistic

-2.0166666666666666

## What significance threshold should we use? 

We actually do not know what significance threshold Pachyderm Adventures used to make their claim of significance! 

**Let's use a significance level of $\alpha = 0.05$.**

## What is the critical value of the test statistic that we should compare our test statistic against? 

Since we are performing a two-tailed one-sample z-test and have set our significance threshold at $\alpha = 0.05$, we would reject the null hypothesis if the test statistic falls within the critical region in the upper or lower tails of a standard normal distribution. 

To achieve a significance level of $\alpha = 0.05$, the critical region in each tail, lower and upper, must have an area under the curve of 0.025. 

The critical values of the z-statistic for a two-sided test annd $\alpha = 0.05$ are $\pm$ 1.96. 

## Is Pachyderm Adventures correct? 

The z-statistic we computed is -2.02 (rounded to two decimal places). This value is less than -1.96, so we have enough evidence to reject the null hypothesis at a significance level of $\alpha = 0.05$. 

(We actually do not know if Pachyderm Adventures is right or not, because they didn't specify the significance threshold they used to make their claim!)

## Another way to compute this: 

In [5]:
from scipy import stats

In [6]:
stats.norm.cdf(z_statistic)

0.021865155367929986

* The area of tail corresponding to the z_statistic we computed is 2.2%. This is less than the threshold of 2.5%, so we have enough evidence to reject the null hypothesis at a significance level of $\alpha = 0.05$. 

# Summary

Key takeaways:

1. 